### PR TITLE
New device fixture factories.

### PR DESCRIPTION
**Pull request by joelwetzel**
This PR separates out the new device fixture factories and their tests, so that they don't have to be in PR https://github.com/biocomp/hubitat_ci/pull/56 anymore.

While doing that, it also changes how the fixtures handle attribute values, to make it match how the real Hubitat does it.  They are no longer stored in device state, but in a separate map of attribute values that can be queried with the device's currentValue() method.

### CODE REVIEW

File: src/main/groovy/me/biocomp/hubitat_ci/util/device_fixtures/LockFixtureFactory.groovy [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        // Calling initialize attaches behavior involving commands, state maintenance, and sending events.
+        lockMetaClass.initialize = { appExecutor, initialAttributeValues ->
+            attributeValues = initialAttributeValues
+
+            // A group of testing-only state variables, that allow us to simulate different aspects of real physical locks in our tests.
+            //  - commandsToIgnore: This allows us to simulate a lock that is unresponsive for a bit.  It will ignore the specified number of commands before responding again.
+            //  - requireRefresh: Some older smart locks don't immediately report back their state changes.  You have to give them a refresh command to get the latest state from them.
+            //  - desiredState: If requireRefresh is set to true, this is the state we want the lock to be in after the refresh command is sent.
+            lockMetaClass.state = [commandsToIgnore: 0, requireRefresh: false, desiredState: initialAttributeValues.lock]
```
Comment (biocomp) [inline src/main/groovy/me/biocomp/hubitat_ci/util/device_fixtures/LockFixtureFactory.groovy:33]:
```suggestion
            lockMetaClass.state = [numOfCommandsToIgnore: 0, requireRefresh: false, desiredState: initialAttributeValues.lock]
```

Perhaps this name provides more context

**Review by biocomp — COMMENTED** [review]
Thanks for separating this out!

**Review by biocomp — APPROVED** [review]

File: src/main/groovy/me/biocomp/hubitat_ci/util/device_fixtures/LockFixtureFactory.groovy (hunk 1 above)
Comment (joelwetzel) [reply src/main/groovy/me/biocomp/hubitat_ci/util/device_fixtures/LockFixtureFactory.groovy]: Good idea.  Done.

**Comment by joelwetzel** [comment]
Ok, I think we're ready to merge.  Thanks for the review!

**Pull request merged**